index.bs

```diff
@@ -2554,12 +2554,45 @@ as it implies that it's possible to dispatch an event asynchronously.
 All events are dispatched synchronously.
 What is more often implied by "asynchronous event" is to defer firing an event.
 
-<h3 id="state-and-subclassing">Use plain {{Event}}s for state</h3>
+<h3 id="state-and-subclassing">Put state on {{Event/target}} objects rather than {{Event}}s</h3>
 
-Where possible, use a plain {{Event}} with a specified `type`,
-and capture any state information in the {{Event/target}} object.
+Where possible, rather than put state on {{Event}} objects, put state on the {{Event/target}}
+and use events to signal updates to that state.
 
-It's usually not necessary to create new subclasses of {{Event}}.
+<div class="note">
+This means the {{Event/target}} object can used to determine the current state,
+without waiting for the next \'update\' event.
+
+This is particularly useful if there's a final state,
+where there will be no further \'update\' events.
+</div>
+
+It's usually not necessary to create new subclasses of {{Event}},
+but they can be used to provide information relating to how the state change occurred.
+
+<div class="example">
+Properties on {{HTMLInputElement}},
+such as {{HTMLInputElement/value}},
+provide the state of the input.
+Properties on {{InputEvent}},
+such as {{InputEvent/inputType}},
+describe the nature of an update to the state.
+</div>
+
+<div class="note">
+An example where this is handled inconsistently is {{XMLHttpRequest}}.
+Some state is on the {{XMLHttpRequest|target}},
+such as {{XMLHttpRequest/readyState}} and {{XMLHttpRequest/status}},
+whereas other state,
+such as {{ProgressEvent/lengthComputable}} and {{ProgressEvent/total}},
+is only available on {{ProgressEvent}} objects via a {{XMLHttpRequestEventTarget/progress}} event.
+
+If the developer gets access to the {{XMLHttpRequest}} object after the first {{XMLHttpRequestEventTarget/progress}} event,
+the current {{ProgressEvent/lengthComputable}} and {{ProgressEvent/total}} values are inaccessible to them.
+
+Instead, all state should have been on the {{XMLHttpRequest}} object,
+and the {{XMLHttpRequestEventTarget/progress}} event should have been a simple {{Event}}.
```

Comment:
```suggestion
```

I tend to think that you can economise here a little and rely on implying the better design, rather than fully spelling it out.